File: 1313-decompress-run-length-encoded-list/1313-decompress-run-length-encoded-list.c
int* decompressRLElist(int* nums, int numsSize, int* returnSize){
    int prev = 0; *returnSize = 0;
    
    for (int i = 0; i < numsSize; i += 2) *returnSize += nums[i];
    int* returnNums = (int*) malloc(sizeof(int) * (*returnSize));
    
    for (int i = 0; i < numsSize; i += 2) for (int j = 0; j < nums[i]; j++) returnNums[prev++] = nums[i + 1];
    return returnNums;
}

File: 1313-decompress-run-length-encoded-list/1313-decompress-run-length-encoded-list.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int size=0,p=0;
        for(int i=0;i<nums.length;i+=2) size+=nums[i];
        int []arr=new int[size];
        for(int i=0;i<nums.length;i+=2)
            for(int j=0;j<nums[i];j++) arr[p++]=nums[i+1];
        return arr;
    }
}